packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/stoppable_timer.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+/// {@template amplify_analytics_pinpoint_dart.stoppable_timer}
+/// A Timer that can be stopped and started again.
+///
+/// Used for auto flush of events
+/// Timer must be stoppable to support the public API's disable method
```

Comment:
I would focus the comment on what the code does, rather than how it will be used internally.

> The doc comment should focus on what the code _does_.
- https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-function-or-method-comments-with-third-person-verbs

```suggestion
```